### PR TITLE
Fix takeattendance bugs

### LINKED ISSUE #177: [PE-D][Tester D] takeattendance command no error message
when invalid input is given to takeattendance, for example a student that does not exist / session that does not exist, no error message is given to the user, may not be intuitive for the user to know what is wrong with the command they entered.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: samuelmui8/ped#4

### LINKED ISSUE #181: [PE-D][Tester B] Bug in Session List
Steps to reproduce
1. Session id 4 does not exist
2. updatesessionremark s/4 r/REMARK

Expected
Error Message as session id 4 does not exist.

Actual
Nothing happens

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: Carlintyj/ped#6

### LINKED ISSUE #186: [PE-D][Tester C] User Guide example error
Marks present for 5th session but the input is for session 2

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: Jonyxzx/ped#2

### LINKED ISSUE #194: [PE-D][Tester C] Unhandled invalid input
There is no error message for invalid input at all

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Jonyxzx/ped#8

### PR DESCRIPTION
**Pull request by ldinghan**

### CODE REVIEW

**Comment by ldinghan** [comment]
Also to provide fix for invalid session number input when updating remark / deleting session

File: src/main/java/seedu/address/logic/commands/TakeAttendanceCommand.java [hunk 1]
```diff
@@ -41,8 +45,9 @@ public class TakeAttendanceCommand extends Command {
 
     public static final CommandType COMMAND_TYPE = CommandType.TAKE_ATTENDANCE;
 
+    public static final String MESSAGE_PERSON_NOT_FOUND = "Not all names given are valid students";
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/commands/TakeAttendanceCommand.java:48]: Might want to change this to 
`MESSAGE_INVALID_STUDENT`

File: src/test/java/seedu/address/logic/commands/UpdateSessionRemarkCommandTest.java [hunk 2]
```diff
@@ -53,6 +55,18 @@ public void execute_validSessionNumberAndSessionRemark_success() throws CommandE
         assertEquals(expectedSession, updatedSession);
     }
 
+    @Test
+    public void execute_invalidSessionNumber_throwsCommandException() {
```
Comment (raydenlim) [inline src/test/java/seedu/address/logic/commands/UpdateSessionRemarkCommandTest.java]: throws `SessionNotFoundException` ? not `CommandException`

File: src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java [hunk 3]
```diff
@@ -90,6 +92,32 @@ public void execute_takeAttendanceWithMultipleStudents_success() throws CommandE
         assertFalse(updatedSession.getStudents().contains(bob));
     }
 
+    @Test
+    public void execute_invalidArguments_throwsCommandException() {
```
Comment (raydenlim) [inline src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java]: throws `SessionNotFoundException` ? not `CommandException`

File: src/test/java/seedu/address/model/session/SessionListTest.java [hunk 4]
```diff
@@ -85,7 +85,8 @@ public void remove_throwsSessionNotFoundException() {
     @Test
     public void find_findSessionByNumber_notInSessionList() {
```
Comment (raydenlim) [inline src/test/java/seedu/address/model/session/SessionListTest.java]: change name of test case to suit the method tested and outcome

File: src/main/java/seedu/address/logic/commands/TakeAttendanceCommand.java [hunk 5]
```diff
@@ -169,9 +173,7 @@ public boolean equals(Object other) {
                     && this.sessionNumber.equals(otherTakeAttendanceCommand.sessionNumber);
         } else if (names == null && otherTakeAttendanceCommand.names == null) {
             // Compare when both 'names' are null
-            return this.name == null ? otherTakeAttendanceCommand.name == null
-                    : this.name.equals(otherTakeAttendanceCommand.name)
-                    && this.sessionNumber.equals(otherTakeAttendanceCommand.sessionNumber);
+            return this.sessionNumber.equals(otherTakeAttendanceCommand.sessionNumber);
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/commands/TakeAttendanceCommand.java:176]: Write a testcase for this :)

**Review by raydenlim — APPROVED** [review]
Other than a few small issue, LGTM!

File: src/test/java/seedu/address/logic/commands/UpdateSessionRemarkCommandTest.java (hunk 2 above)
Comment (ldinghan) [reply src/test/java/seedu/address/logic/commands/UpdateSessionRemarkCommandTest.java]: youre right

File: src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java (hunk 3 above)
Comment (ldinghan) [reply src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java]: youre right, i split the test into two as well

**Pull request merged**